### PR TITLE
#34 Whitelist XML root elements

### PR DESCRIPTION
**Pull request by kaklakariada**
Skip files with unknown root element. See #34

### CODE REVIEW

**Comment by kaklakariada** [comment]
Codecov/changes fails with status "14 files have unexpected coverage changes not found in diff."
No idea what this means, I would ignore it.

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java [hunk 1]
```diff
@@ -55,36 +67,73 @@ void runImport() throws XMLStreamException
...
+                if (skipFile(currentEvent.asStartElement()))
+                {
+                    return;
+                }
                 foundStartElement(currentEvent.asStartElement());
             }
         }
     }
 
+    private boolean skipFile(final StartElement startElement)
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: The name "skipFile" does not fit. No file is skipped in this method. Instead this method checks the conditions that lead to skipping a file in another method.

Suggested: "isXmlStartWhitelisted()"

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java [hunk 2]
```diff
@@ -55,36 +67,73 @@ void runImport() throws XMLStreamException
...
+    {
+        if (this.validRootElementFound)
+        {
+            return false;
+        }
+        if (rootElementWhitelisted(startElement))
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: verb missing in method name.

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java [hunk 3]
```diff
@@ -55,36 +67,73 @@ void runImport() throws XMLStreamException
...
+
+    private boolean rootElementWhitelisted(final StartElement rootElement)
+    {
+        final String elementName = rootElement.getName().getLocalPart();
+        return SPECDOCUMENT_ROOT_ELEMENT_NAME.equals(elementName);
+    }
+
     private void foundStartElement(final StartElement element) throws XMLStreamException
     {
         switch (element.getName().getLocalPart())
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: Code duplication with line 97. Better pass around element name instead?

**Comment by redcatbear** [comment]
Only small findings. Can be merged after correction.

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java (hunk 3 above)
Comment (kaklakariada) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: We need the StartElement later in `foundStartElement()` to get attributes from it. I would prefer not to add an parameter for the element name with duplicate information.

**Comment by kaklakariada** [comment]
Thank you for the review, I will merge the PR.

**Pull request merged**